libs/server/src/lib/api/executable.ts

```diff
@@ -0,0 +1,160 @@
+import { normalizeCommands } from '../utils/architect';
+import { FileUtils } from '../utils/file-utils';
+import { readJsonFile } from '../utils/utils';
+import { Commands } from './commands';
+import { createDetailedStatusCalculator } from './detailed-status-calculator';
+import { Telemetry } from '../telemetry';
+
+// todo(matt) should be handled by Commands
+let commandRunIndex = 0;
+
+export interface RunResult {
+  id: string;
+}
+
+export enum ExecutableType {
+  add = 'add',
+  new = 'new',
+  generate = 'generate',
+  npm = 'npm',
+  ng = 'ng'
+}
+
+// TODO(matt): next step of this refactoring is 1 level of inheritance to
+// handle differences in name / path behaviour for different executables
+export class Executable {
+  private readonly isNvm: boolean;
+
+  private _path = '';
+  private _name: string;
+
+  get name() {
+    return this.isNvm ? 'nvm' : this._name;
+  }
+
+  get path() {
+    return this.isNvm ? `nvm exec ${this._path}` : this._path;
+  }
+
+  setPath(path: string) {
+    this._path = path;
+  }
+
+  setName(name: string) {
```

Comment:
You already have a getter, why not just have a setting too?

Reply:
I get this feeling someday someone is going to just try to set the property and be really consfused when nothing happens

Reply:
you are totally right